apps/studio/prisma/schema.prisma

```diff
@@ -221,3 +222,32 @@ model Whitelist {
 
   @@index([email])
 }
+
+enum AuditLogEvent {
+  ResourceCreate
+  ResourceUpdate
+  ResourceDelete
+  ResourceMove
+  UserCreate
+  UserUpdate
+  UserDelete
+  Publish
+  Login
+  Logout
+  PermissionCreate
+  PermissionUpdate
+  PermissionDelete
+}
+
+model AuditLog {
+  id        Int           @id @default(autoincrement())
+  user      User?         @relation(fields: [userId], references: [id])
+  userId    String?
+  eventType AuditLogEvent
+  createdAt DateTime      @default(now())
+  metadata  Json
+  /// @kyselyType(PrismaJson.AuditLogJsonContent)
+  /// [AuditLogJsonContent]
+  delta     Json
+  ipAddress String?       @db.Inet
+}
```

Comment:
- Should we have a `updatedAt` table as another proof that this record has not been tampered with _(proven as long as `createdAt` = `updatedAt`)_?
- `user` and `userId` should not be optional since there should always be a user tied to the action

Reply:
done both, good points!